wilds/datasets/py150_dataset.py

```diff
@@ -81,15 +83,23 @@ def __init__(self, version=None, root_dir='data', download=False, split_scheme='
         self._y_size = None
 
         _repo = torch.tensor(df['repo'].values).reshape(-1,1)  #[n_samples, 1]
-        _mask = torch.tensor(list(df['mask'].apply(lambda x: x[1:]).values)) #[n_samples, seqlen-1]
-        self._metadata_array = _repo
-        self._metadata_fields = ['repo']
+        _tok_type = torch.tensor(list(df['tok_type'].apply(lambda x: x[1:]).values)) #[n_samples, seqlen-1]
+        length = _tok_type.size(1)
+        self._metadata_fields = ['repo'] + [f'tok_{i+1}_type' for i in range(length)]
+        self._metadata_array = torch.cat([_repo, _tok_type], dim=1)
 
         self._y_array = self._y_array.float()
-        self._y_array[(1-_mask).bool()] = float('nan')
+        self._y_array[(_tok_type==-100).bool()] = float('nan')
 
         super().__init__(root_dir, download, split_scheme)
 
+    def _compute_acc(self, y_pred, y_true, eval_pos):
+        flattened_y_pred = y_pred[eval_pos]
+        flattened_y_true = y_true[eval_pos]
+        assert flattened_y_pred.size()==flattened_y_true.size() and flattened_y_pred.dim()==1
+        acc = (flattened_y_pred==flattened_y_true).sum() / (len(flattened_y_pred) +1e-8)
+        return acc

```

Comment:
Is acc a tensor? In that case, can you turn it into a plain number (acc.item()) for consistency / proper logging? Thanks!

Reply:
If we're changing this, could we maybe also change it something like this to avoid the very slight distortion?

```python
if len(flattened_y_pred) == 0:
  acc = 0
else:
  acc = (flattened_y_pred==flattened_y_true).mean()
```